schematic/store/synapse.py

```diff
@@ -51,9 +51,98 @@
+        """
+        if CONFIG["synapse"]["manifest_folder"]:
+            manifest_data = syn.get(
+                    manifest_id,
+                    downloadLocation=CONFIG["synapse"]["manifest_folder"],
+                    ifcollision="overwrite.local",
+                )
+        else:
+            manifest_data = syn.get(
+                        manifest_id,
+                    )
+        return manifest_data 
+
+    def _entity_type_checking(self, syn, manifest_id):
+        """
+        check the entity type of the id that needs to be downloaded
+        Args:
+            syn: an object of type synapseclient.
+            manifest_id: id of a manifest
+        Return: 
+             if the entity type is wrong, raise an error
+        """
+        # check the type of entity
+        entity_type = entity_type_checking(syn, manifest_id)
+        if entity_type  != "file":
+            logger.error(f'You are using entity type: {entity_type}. Please try using a file')
+
+    @staticmethod
+    def download_manifest(self, syn, manifest_id: str, newManifestName="", manifest_df=None):
+        """
+        Donwload a manifest based on a given manifest id. 
+        Args:
+            syn: an object of type synapseclient.
+            manifest_id: id of a manifest
+            newManifestName(optional): new name of a manifest that gets downloaded.
+            manifest_df(optional): a dataframe containing name and id of manifests in a given asset view
+        Return: 
+            manifest_data: synapse entity file object
+        """
+
+        # enables retrying if user does not have access to uncensored manifest
+        # pass synID to synapseclient.Synapse.get() method to download (and overwrite) file to a location
+        manifest_data = ""
+
+        # check entity type
+        self._entity_type_checking(syn, manifest_id)
+
+        # download a manifest
+        try:
+            manifest_data = self._download_manifest_to_folder(syn, manifest_id)
+        except(SynapseUnmetAccessRestrictions, SynapseAuthenticationError):
+            if not manifest_df.empty:
+                censored_regex=re.compile('.*censored.*')
+                censored = manifest_df['name'].str.contains(censored_regex)
+                new_manifest_id=manifest_df[censored]["id"][0]
+                try: 
+                    manifest_data = self._download_manifest_to_folder(syn, new_manifest_id)
+                except (SynapseUnmetAccessRestrictions, SynapseAuthenticationError) as e:
+                    logger.error(f"You don't have access to the requested resource: {manifest_id}")
+                    raise (f"You don't have access to the requested resource: {manifest_id}")
+
+            else:
+                logger.error(f"You don't have access to the requested resource: {manifest_id}")
+
+        if newManifestName:
+            if os.path.exists(manifest_data['path']):
```

Comment:
I think this can be condensed into one line:

```
if newManifestName and os.path....
```

Reply:
I like what you suggested. Thank you Tom.

Reply:
So what is the impact if this is an empty string.  By default if it is an empty string os.path.exists is going to be False.

Now if what you mean by doesnt exist is that there will be a KeyError, then you can do manifest_data.get('path')

Reply:
I think we could just do `manifest_data.get('path')` like you suggested to avoid key error. Thank you!